UI/src/main.tsx

```diff
@@ -0,0 +1,87 @@
+import { Navigate, Route, Routes } from "react-router-dom";
+import { useToast } from "@chakra-ui/react";
+import Header from "./components/header/view";
+import Unauthorised from "./unauthorised/view";
+import Footer from "./components/footer/footer";
+import { useTranslate } from "./translate";
+import * as lang from "./translations/basicTranslationSet";
+import { useSelector } from "react-redux";
+import { Selectors as menuSelectors } from "./redux/modules/menu";
+import { Selectors as errorSelectors } from "./redux/modules/errorHandler";
+import {
+  notificationResetNotificationAction,
+  Selectors as notificationSelector,
+} from "./redux/modules/common/notification";
+
+import AuthLoader from "./authLoader/view";
+
+import { Box, Container } from "@chakra-ui/react";
+
+import SimpleSidebar from "./components/sideBar/sideBar";
+import { mainSections } from "./siteMap/mainSections";
+
+import { useDispatch } from "react-redux";
+
+import { ChackraUIBaseModal } from "./componentsCommon/chackraModal/chackraBaseModal";
+import { ProblemDetailsModal } from "./componentsCommon/problemDetailsModal/problemDetailsModal";
+
+const Component = () => {
+  const translate = useTranslate();
+  const dispatch = useDispatch();
+  return (
+    <>
+      <Routes>
+        <Route index path="/" element={<Navigate to="/edm/manualUpload" />} />
+        <Route path="/Unauthorized" element={<Unauthorised />} />
+        <Route path="/main/Test/" element={<></>} />
+
+        {
+          //Understand why with this method :params in path are not being recognized.
+          /*mainSections.map((x) =>
+          x.subsections.map((s) => (
+            <Route path={s.path} element={s.component({ a: dispatch })} />
+          ))
+        )*/
+        }
+      </Routes>
+    </>
+  );
+};
+
+const Main = () => {
+  const userRole = useSelector(menuSelectors.userRole);
+  const problemDetails = useSelector(errorSelectors.all);
+  const notification = useSelector(notificationSelector.all).notification;
+  const dispatch = useDispatch();
+  var toast = useToast();
+  return userRole === "" ? (
```

Comment:
this doesn't look sane. the "Login" should not depend on the Logged User to have a Role which actually comes from the Menu system ...